### PR TITLE
feat(build): --push and --containerize

### PR DESCRIPTION
**Pull request by aarnphm**
note that these two options are mutually exclusive

Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml_cli/bentos.py [hunk 1]
```diff
@@ -339,29 +363,60 @@ def build(  # type: ignore (not accessed)
...
-            click.secho(f"Successfully built {bento}.", fg="green")
-
-            click.secho(
-                f"\nPossible next steps:\n\n * Containerize your Bento with `bentoml containerize`:\n    $ bentoml containerize {bento.tag}",
-                fg="blue",
+        if machine:
+            click.echo(f"__tag__:{bento.tag}")
+        elif output == "tag":
+            click.echo(bento.tag)
+        elif output == "json":
```
Comment (ssheng) [inline src/bentoml_cli/bentos.py:375]: What's the JSON output for?

Comment (aarnphm) [reply src/bentoml_cli/bentos.py:375]: I don't think we need json here, I was playing with a CLI use case for a prompt this weekend. I will remove it for now.

File: src/bentoml/bentos.py [hunk 2]
```diff
@@ -360,15 +359,15 @@ def build(
...
+    build_args = ["bentoml", "build", "--machine"]
 
     if build_ctx is None:
         build_ctx = "."
     build_args.append(build_ctx)
 
     if version is not None:
         build_args.extend(["--version", version])
-    build_args.extend(["--output", "tag"])
+    build_args.extend(["--machine"])
```
[the remark this answers is not in the document]
Comment (ssheng) [reply src/bentoml/bentos.py:370]: Can we make `--output tag` to behave the way you wished without introduce another option in `--machine`?

[the remark this answers is not in the document]
Comment (aarnphm) [reply src/bentoml/bentos.py:370]: `--output tag` before just prints out the tag, and I think it doesn't make sense to have `--output tag` to have `__tag__:<tag>`.

`--machine` will be internal and hidden, so `bentoml build -h` won't show `--machine`

**Review by ssheng — APPROVED** [review]

**Comment by aarnphm** [comment]
CI will be addressed in #4056

**Pull request merged**